src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -800,9 +983,22 @@ void Graph::AllocateWithReuse(const std::vector<size_t>& syncNodesInds) {
 
         int64_t boxSize = 0;
         bool isConst = false, isOutput = false, isInput = false;
-        for (auto &edge : edge_clusters[i]) {
-            int e_start = edge->getParent()->getExecIndex();
-            int e_finish = edge->getChild()->getExecIndex();
+
+        for (auto &edge : clusters[i]) {
+            const auto& parent = edge->getParent();
+            const auto& child = edge->getChild();
+
+            auto usesInOutMemoryMultipleTimes = [](const NodePtr& node) {
+                if (auto tensorIterator = std::dynamic_pointer_cast<node::TensorIterator>(node)) {
+                    return tensorIterator->usesInOutMemoryMultipleTimes();
+                }
```

Comment:
Can we slightly speed up check by testing the `getType` value first and call dync cast only if the type is  `Type::TensorIterator`?

Reply:
Done